### DISCUSSION TITLE
Can't Get it to Work on Digital Ocean

### DISCUSSION THREAD

**Discussion by osseonews** [vendurehq/vendure#666] [General]
Tried to install this on DO's basic NodeJS app (See: https://marketplace.digitalocean.com/apps/nodejs ) 

Application installs, but can't get it work after following all the instructions. Site never loads. 

If you can just provide a basic step-by-step explanation of how to install and get this live on the DO's Nodejs marketplace app installation would be very helpful. I'm sure it's really easy, and I'm missing some step. DO itself provides a good explanation in the link above how to get an app running and we can do it with other apps, but can't get Vendure to run for some reason.

Thanks.

**Comment by michaelbromley** [comment]
> can't get it work after following all the instructions. Site never loads

Can you provide more exact information? Any errors logged? etc

**Comment by osseonews** [comment]
It just won't run. There aren't any specific errors. Just can't access anything at the URLs, like 3000:/admin (not sure I'm typing this correctly). It never loads at the IP address. Maybe the file we are running is wrong? We tried to run index.js in PM2. Probably something to do with PM2 or Ngnix settings when trying to map to the app. I'm not sure, as I'm not clear as to what exactly has to be running.

Anyway, if you can just provide a step-by-step instructions on how to get this running on DO's Nodejs marketplace app, based on the link above to DO set up, that would be helpful. You can set it up for $5 and see just document how you did it. This is the main marketplace app for Nodejs and it is extremely popular for running any node app. So just giving instructions on how to get this up and running with that set up would do alot for Vendure. I'm sure it is very easy, but it's not clear from any of the instructions on how to actually get this live on a real server. Getting Vendure running locally is easy, but obviously not a long-term goal.

Thanks for all your help.

**Comment by michaelbromley** [comment]
I would start by SSHing into the DO server and check the logs that should be output by Vendure. From inside the server you can also use curl to localhost:3000 to see if the server is actually running.

It might also be an nginx config issue, but I don't know your nginx config.

> Anyway, if you can just provide a step-by-step instructions on how to get this running on DO's Nodejs marketplace app, based on the link above to DO set up, that would be helpful

I would love to provide resources like this. Right now I don't have the time as I am working on getting the 1.0 release done. In future I will dedicate more time to such resources and learning materials.

**Comment by osseonews** [comment]
OK. Thanks. Can you just confirm that index.js is the correct file to be running for PM2? For continuous deployment of a nodejs app, you pretty much need PM2.

**Reply by michaelbromley** [reply]
Yes, index.js and also index-worker.js needs to be running.
